docs/sandbox-permissions.rst

```diff
@@ -223,9 +223,88 @@ requested with ``--filesystem`` and are not available with
 
 Device access
 `````````````
+You can provide the following device permissions:
+
+========= ======================================================
+``dri``   Direct Rendering Interface. Necessary for GL.
+``kvm``   Kernel based Virtual Machine ``/dev/kvm``
+``shm``   Shared Memory in ``/dev/shm``.
+``input`` Input devices as exposed in ``/dev/input``. This includes game controllers. Since Flatpak 1.15.6.
+``all``   All devices, including all of the above except ``shm``
+========= ======================================================
+
+.. note::
+
+   ALSA sound devices in ``/dev/snd`` are exposed with the socket
+   permission ``pulseaudio``.
 
 While not ideal, ``--device=all`` can be used to access devices like
-controllers or webcams.
+generic USB or webcams.
+
+Newer permissions like ``input``, meant to avoid using ``all`` will
+not work on older versions. Since not all distributions or users might
+be running latest version of Flatpak supporting this feature, it is
+needed to be able to be compatible with older versions.
+
+Starting with Flatpak 1.16, there is a device fallback mechanism. The
+device fallback mechanism is meant to specify that a certain device
+permission is to be used with a fallback, and therefor, if Flatpak
+knows about it, it can tighten device access as intended and ignore
+the fallback.
+
+Packagers should take great care in considering device permissions and
+use the fallback mechanism when available.
+
+For example in the case where input device access is need you could
+specify:
+
+.. code-block::
+
+   --device=input
+
+.. warning::
+
+   On the command line, with Flatpak < 1.15.6, you will get an
+   error. In the metadata, it will be ignored making the package not
+   function as intended and offer a poor user experience.
+
+The solution is to use a fallback.
+
+Fallback gets specified with the ``--device`` option like for device
+permissions. It starts with ``fallback:`` and then list the
+permission, followed by the fallback, separated by a comma. In most
+case the fallback will be ``all`` as there is currently no overlap
+otherwise.
+
+The generic syntax is:
+
+.. code-block::
+
+   --device=fallback:DEVICE_PERMISSION:FALLBACK_PERMISSION
+   --device=FALLBACK_PERMISSION
+
+Specifying a fallback permission (second line) is required.
+
+In the example above, the fallback syntax would be as follow:
+
+.. code-block::
+
+  --device=fallback:input,all
+  --device=all
+
+This translates to "grant permission to ``input`` devices if Flatpak
+supports it, otherwise fallback to granting ``all`` device access".
```

Comment:
```suggestion
This translates to "grant permission to ``input`` devices if Flatpak
supports it and ignore ``all``, otherwise fallback to granting ``all`` device access".
```